src/overlays/ovl_i4/fox_sz.c

```diff
@@ -1,55 +1,1853 @@
+        }
+
+        if ((D_i4_801A0560 == 3) && (D_800D3180[8] != 0)) {
+            D_800C9B4C = D_8015F928 + 110;
+        }
+    }
+}
+
+void func_i4_80199B18(Effect* effect, f32 xPos, f32 yPos, f32 zPos, f32 xVel, f32 yVel, f32 zVel, f32 scale) {
+    Effect_Initialize(effect);
+    effect->obj.status = 1;
+    effect->obj.id = OBJ_EFFECT_339;
+
+    effect->obj.pos.x = xPos;
+    effect->obj.pos.y = yPos;
+    effect->obj.pos.z = zPos;
+
+    effect->vel.x = xVel;
+    effect->vel.y = yVel;
+    effect->vel.z = zVel;
+
+    effect->scale2 = scale;
+    effect->scale1 = 0.5f;
+
+    effect->unk_4C = 2;
+    effect->obj.rot.z = Rand_ZeroOne() * 360.0f;
+
+    Object_SetInfo(&effect->info, effect->obj.id);
+
+    effect->unk_44 = 255;
+    effect->unk_46 = 2;
+}
+
+void func_i4_80199BDC(f32 xPos, f32 yPos, f32 zPos, f32 xVel, f32 yVel, f32 zVel, f32 scale) {
+    s32 i;
+
+    for (i = ARRAY_COUNT(gEffects) - 1; i >= 0; i--) {
+        if (gEffects[i].obj.status == 0) {
+            func_i4_80199B18(&gEffects[i], xPos, yPos, zPos, xVel, yVel, zVel, scale);
+            break;
+        }
+    }
+}
+
+// https://decomp.me/scratch/FVgio
+#ifdef NON_MATCHING
+void func_i4_80199C60(Actor* actor) {
+    s32 pad[2];
+    f32 var_fa0;
+    f32 var_fv1;
+
+    switch (++actor->iwork[9]) {
+        case 600:
+            Radio_PlayMessage(gMsg_ID_16080, RCID_ROB64);
+            break;
+
+        case 1000:
+            Radio_PlayMessage(gMsg_ID_16085, RCID_ROB64);
+            break;
+
+        case 1200:
+            Radio_PlayMessage(gMsg_ID_16090, RCID_ROB64);
+            break;
+    }
+
+    var_fv1 = gBosses->obj.pos.x - actor->obj.pos.x;
+    var_fa0 = gBosses->obj.pos.z - actor->obj.pos.z;
+    __sinf((gGameFrameCount * M_DTOR));
+
+    if (0) {
+        ;
+    }
+    if (var_fv1) {
+        ;
+    }
+    if (var_fa0) {
+        ;
+    }
+
+    if (actor->unk_0E4 < 100) {
+        var_fv1 = __sinf(((f32) ((actor->index * 45) + gGameFrameCount)) * M_DTOR) * 5000.0f;
+        var_fa0 = __cosf(((f32) ((actor->index * 45) + (gGameFrameCount * 2))) * M_DTOR) * 5000.0f;
+    } else {
+        var_fa0 = 0.0f;
+        var_fv1 = 0.0f;
+    }
+
+    actor->fwork[4] = (gBosses->obj.pos.x + var_fv1) + 400.0f;
+    actor->fwork[5] = (gBosses->obj.pos.y + var_fa0) + 100.0f;
+    // temp_fv0 = fabsf(actor->fwork[4] - actor->obj.pos.x);
+    actor->fwork[6] = gBosses->obj.pos.z;
+    actor->fwork[3] = 1.4f;
+
+    if (((fabsf(actor->fwork[4] - actor->obj.pos.x) < 800.0f) &&
+         (fabsf(actor->fwork[5] - actor->obj.pos.y) < 800.0f)) &&
+        (fabsf(actor->fwork[6] - actor->obj.pos.z) < 800.0f)) {
+        func_i4_80199900(actor, 0);
+        D_80178480 = 0x19;
+        gBosses->dmgType = 100;
+        if ((gPlayer->state_1C8 == PLAYERSTATE_1C8_3) || (gPlayer->state_1C8 == PLAYERSTATE_1C8_5)) {
+            gPlayer->state_1C8 = PLAYERSTATE_1C8_7;
+            gPlayer->unk_1D0 = 0;
+            gActors->state = -0x7960;
```

Comment:
```suggestion
            gActors->state = -31072;
```
I prefer states to be in decimal, although what a strange number to choose.